bgpd/bgp_route.c

```diff
@@ -3901,10 +3901,10 @@ bgp_static_update (struct bgp *bgp, struct prefix *p,
 
 	  /* Nexthop reachability check. */
          if (bgp_flag_check (bgp, BGP_FLAG_IMPORT_CHECK) &&
-              safi == SAFI_UNICAST)
+             (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
 	    {
 	      if (bgp_find_or_add_nexthop (bgp, afi, ri, NULL, 0) &&
-                  safi == SAFI_UNICAST)
+                  (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
```

Comment:
Agree about the unneeded test. I modified the existing code not noticing that the second test was really unnecessary.  I will also check the later check BGP_FLAG_IMPORT.  Thanks for the review